src/EnergyPlus/CurveManager.cc

```diff
@@ -442,60 +438,62 @@ namespace CurveManager {
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff9 = Numbers(9);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff10 = Numbers(10);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff11 = Numbers(11);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff12 = Numbers(12);
-
-            state.dataCurveManager->PerfCurve(CurveNum).Var1Min = Numbers(13);
-            state.dataCurveManager->PerfCurve(CurveNum).Var1Max = Numbers(14);
-            state.dataCurveManager->PerfCurve(CurveNum).Var2Min = Numbers(15);
-            state.dataCurveManager->PerfCurve(CurveNum).Var2Max = Numbers(16);
-            state.dataCurveManager->PerfCurve(CurveNum).Var3Min = Numbers(17);
-            state.dataCurveManager->PerfCurve(CurveNum).Var3Max = Numbers(18);
+            auto &thisCurve = state.dataCurveManager->PerfCurve(CurveNum);
+
+            thisCurve.Name = Alphas(1);
+
+            thisCurve.curveType = CurveType::ChillerPartLoadWithLift;
+            thisCurve.ObjectType = CurrentModuleObject;
+            thisCurve.NumDims = 3;
+            thisCurve.InterpolationType = InterpType::EvaluateCurveToLimits;
+
+            thisCurve.coeff[0] = Numbers(1);
+            thisCurve.coeff[1] = Numbers(2);
+            thisCurve.coeff[2] = Numbers(3);
+            thisCurve.coeff[3] = Numbers(4);
+            thisCurve.coeff[4] = Numbers(5);
+            thisCurve.coeff[5] = Numbers(6);
+            thisCurve.coeff[6] = Numbers(7);
+            thisCurve.coeff[7] = Numbers(8);
+            thisCurve.coeff[8] = Numbers(9);
+            thisCurve.coeff[9] = Numbers(10);
+            thisCurve.coeff[10] = Numbers(11);
+            thisCurve.coeff[11] = Numbers(12);
+
+            thisCurve.inputLimits[0].min = Numbers(13);
+            thisCurve.inputLimits[0].max = Numbers(14);
+            thisCurve.inputLimits[1].min = Numbers(15);
+            thisCurve.inputLimits[1].max = Numbers(16);
+            thisCurve.inputLimits[2].min = Numbers(17);
+            thisCurve.inputLimits[2].max = Numbers(18);
 
             if (NumNumbers > 18 && !state.dataIPShortCut->lNumericFieldBlanks(19)) {
-                state.dataCurveManager->PerfCurve(CurveNum).CurveMin = Numbers(19);
-                state.dataCurveManager->PerfCurve(CurveNum).CurveMinPresent = true;
+                thisCurve.outputLimits.min = Numbers(19);
+                thisCurve.outputLimits.minPresent = true;
             }
             if (NumNumbers > 19 && !state.dataIPShortCut->lNumericFieldBlanks(20)) {
-                state.dataCurveManager->PerfCurve(CurveNum).CurveMax = Numbers(20);
-                state.dataCurveManager->PerfCurve(CurveNum).CurveMaxPresent = true;
+                thisCurve.outputLimits.max = Numbers(20);
+                thisCurve.outputLimits.maxPresent = true;
             }
 
             if (NumAlphas >= 2) {
                 if (!IsCurveInputTypeValid(Alphas(2))) {
```

Comment:
This should be a `getEnumerationValue`.

Reply:
:+1:

Reply:
Hmm.  @JasonGlazer 

Do we just literally not do _anything_ with the unit types of the curves inside E+?  Are they purely tags on the input objects to allow third parties to pass data through?  I think my tone might sound a little antagonistic, but I assure you it's just piqued my curiosity.  Assuming it is, then why do we worry about validating it here?  Just let the user pass whatever they want through.  

If I'm wrong, and we are actually using it, then that's fine, I just don't see it being used except to check the validity.

Reply:
@Myoldmopar I know that IDF Editor will do something with the unit types for input and output but I don't remember if they get used by EnergyPlus. IDF Editor with do a unit conversion for IP for the min and max values, it does not mess with the coefficient. 

Seems like it should be usable by EnergyPlus to validate that the specified curve input object is the right kind when it is getting used. If the algorithm is expecting temperature inputs for X and Y and power output, then that could be validated against. From your question, I guess that is not done but maybe it should be.